.github/workflows/cmake.yml

```
@@ -1,19 +1,33 @@
-    branches: [ master ]
-    tags:
-      - 'v*'
-  pull_request:
+  pull_request_target:
     branches: [ master ]
+    types: [opened, synchronize, reopened]
+  workflow_dispatch:
+
+# Prevent concurrent workflow runs on the same PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ matrix.os }}-${{ matrix.arch }}
```

actionlint

context "matrix" is not allowed here. available contexts are "github", "inputs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)